### ISSUE TITLE
Crash in Release Builds 🆘 🆘 

### ISSUE THREAD

**Issue by Nandha-d3v** [cuvent/react-native-vision-camera#2097]
### What's happening?

The camera worked fine when using in development mode. I was even able to scan couple of Barcodes and do everything as expected. But the moment I launch the camera in Release Build it crashes

"react-native-vision-camera": "^3.6.4",
And I have "react-native-worklets-core": "^0.2.2",
P.S I'm using "@mgcrea/vision-camera-barcode-scanner": "^0.6.3", for scanning barcodes

### Reproduceable Code

```tsx
import {Worklets} from 'react-native-worklets-core';
import {
  Camera,
  useCameraDevice,
} from 'react-native-vision-camera';
import {
  CameraHighlights,
  useBarcodeScanner,
} from '@mgcrea/vision-camera-barcode-scanner';

======>
  const device = useCameraDevice('back');

  const assignBarCodesJs = Worklets.createRunInJsFn(assignBarCodes);

  const {props: cameraProps, highlights} = useBarcodeScanner({
    fps: 2,
    barcodeTypes: [
      'ean-13',
      'code-128',
      'code-39',
      'code-93',
      'codabar',
      'ean-8',
      'itf',
      'upc-e',
    ], // optional
    //scanMode: "once",
    regionOfInterest: {
      x: 0,
      y: 0.5,
      width: 1,
      height: 0.2,
    },
    onBarcodeScanned: codes => {
      'worklet';
      if (codes.length) {
        assignBarCodesJs(codes);
      }
    },
  });

===>Camera component ====>>>
      <Camera
       style={StyleSheet.absoluteFill}
       device={device}
       isActive={keepScanning}
       audio={false}
       {...cameraProps}
       />
```

### Relevant log output

```shell
--------- beginning of crash
10-27 22:35:16.874 32037 32283 E AndroidRuntime: FATAL EXCEPTION: mrousavy/VisionCamera.video
10-27 22:35:16.874 32037 32283 E AndroidRuntime: Process: com.myapp, PID: 32037
10-27 22:35:16.874 32037 32283 E AndroidRuntime: com.facebook.jni.CppException: Compiling JS failed: 1:1:invalid empty parentheses '( )' Buffer size 3 starts with: 280a29
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at com.mrousavy.camera.frameprocessor.FrameProcessor.call(Native Method)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at com.mrousavy.camera.core.VideoPipeline._init_$lambda$0(VideoPipeline.kt:111)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at com.mrousavy.camera.core.VideoPipeline.$r8$lambda$EMriwS_FKhtjTFtppM84Z5V8tiI(Unknown Source:0)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at com.mrousavy.camera.core.VideoPipeline$$ExternalSyntheticLambda0.onImageAvailable(Unknown Source:2)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at android.media.ImageReader$1.run(ImageReader.java:916)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:942)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:226)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:313)
10-27 22:35:16.874 32037 32283 E AndroidRuntime:        at android.os.HandlerThread.run(HandlerThread.java:67)
10-27 22:35:16.884 32037 32283 I Process : Sending signal. PID: 32037 SIG: 9
```

### Camera Device

```json
"sensorOrientation": "landscape-right",
  "hardwareLevel": "full",
  "maxZoom": 10,
  "minZoom": 0.6000000238418579,
  "supportsLowLightBoost": true,
  "neutralZoom": 1,
  "physicalDevices": [
    "ultra-wide-angle-camera",
    "wide-angle-camera",
    "telephoto-camera"
  ],
  "supportsFocus": true,
  "supportsRawCapture": true,
  "isMultiCam": true,
  "name": "BACK (0)",
  "hasFlash": true,
  "hasTorch": true,
  "position": "back",
  "id": "0"
```

### Device

iPhone 13 Pro Max, iPhone 15, Samsung Galaaxy S22

### VisionCamera Version

3.6.4

### Can you reproduce this issue in the VisionCamera Example app?

No, I cannot reproduce the issue in the Example app

### Additional information

- [ ] I am using Expo
- [X] I have enabled Frame Processors (react-native-worklets-core)
- [X] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

**Comment by zzz08900** [comment]
Does it crash on both Android and iOS?
I'm curious if there's any relevant logs from iOS that you can share? 

As for Android, do you have reanimated included in your package.json?

**Comment by mrousavy** [comment]
☝️

**Comment by Nandha-d3v** [comment]
> Does it crash on both Android and iOS? I'm curious if there's any relevant logs from iOS that you can share?
> 
> As for Android, do you have reanimated included in your package.json?

Yes. The crash happens in both Android & iOS
And yes we have "react-native-reanimated": "^3.5.4", 
On the iOS the logs didn't show up anything regarding the crash. But the crash still happens the same way as it did on Android

**Comment by zzz08900** [comment]
Oh yes you can't get crash logs of release builds from iOS without going through some Apple magic.

So for Android the (most likely) problem is `react-native-worklets-core` does not play well with reanimated. There are a few possible ways of fixing this:
1. try using latest `react-native-worklets-core` from Github, I remember there's a fix(by @chrfalch) for this problem but haven't get released yet - be ware you may run into other issues since you will be using the master branch with latest commits included.
2. remove reanimated and refactor all animation to RN's Animated component.
3. some versions of reanimated do work with some versions `react-native-worklets-core`, try and find a nice combination, I would suggest 3.4.x and 0.2.2

Let me know if any of the methods above fixed your crash.

**Comment by chrfalch** [comment]
I'm working on this issue as we speak. Have found the source of the error and will try to push a fix asap.

**Comment by chrfalch** [comment]
This should be fixed in `react-native-workklets-core` now - we just need to upgrade to a new version (just as @zzz08900 said)

**Comment by chrfalch** [comment]
Should be fixed now with version [0.2.4](https://github.com/margelo/react-native-worklets-core/releases/tag/v0.2.4) of `react-native-worklets-core`.

‼️ Please update ‼️

**Comment by Nandha-d3v** [comment]
Now I'm not even able to run the app after updating to  "react-native-worklets-core": "^0.2.4".  

Here is what it says when I try to run it on android

`FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-vision-camera:buildCMakeDebug[arm64-v8a]'.
> com.android.ide.common.process.ProcessException: ninja: Entering directory `/Users/me/myapp/node_modules/react-native-vision-camera/android/.cxx/Debug/1a5r82r4/arm64-v8a'
  [1/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/OpenGLRenderer.cpp.o
  [2/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/PassThroughShader.cpp.o
  [3/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/OpenGLContext.cpp.o
  [4/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/frameprocessor/java-bindings/JFrameProcessorPlugin.cpp.o
  [5/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/frameprocessor/FrameProcessorPluginHostObject.cpp.o
  [6/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/frameprocessor/java-bindings/JFrame.cpp.o
  [7/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/VisionCamera.cpp.o
  [8/16] Building CXX object CMakeFiles/VisionCamera.dir/src/main/cpp/frameprocessor/FrameHostObject.cpp.o
....
....
....

 ld: error: undefined symbol: RNWorklet::JsiHostObject::getPropertyNames(facebook::jsi::Runtime&)
  >>> referenced by JVisionCameraProxy.cpp
  >>>               CMakeFiles/VisionCamera.dir/src/main/cpp/frameprocessor/java-bindings/JVisionCameraProxy.cpp.o:(vtable for RNWorklet::JsiWorklet)
  
  ld: error: undefined symbol: typeinfo for RNWorklet::JsiHostObject
  >>> referenced by JVisionCameraProxy.cpp
  >>>               CMakeFiles/VisionCamera.dir/src/main/cpp/frameprocessor/java-bindings/JVisionCameraProxy.cpp.o:(typeinfo for RNWorklet::JsiWorklet)
  clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
  
  C++ build system [build] failed while executing:
      /Users/me/Library/Android/sdk/cmake/3.22.1/bin/ninja \
        -C \
        /Users/me/myapp/node_modules/react-native-vision-camera/android/.cxx/Debug/1a5r82r4/arm64-v8a \
        VisionCamera
    from /Users/me/myapp/node_modules/react-native-vision-camera/android
`

**Comment by zzz08900** [comment]
Try deleting your node_modules and re-install all dependencies after upgrading worklets-core and see if that fixes your problem.

**Comment by Nandha-d3v** [comment]
Tried that multiple times.It didnt work. It worked only after I cloned the project from start again and then upgraded "react-native-worklets-core" to 0.2.4. I had to do this multiple times whenever I tried to gradle clean. But this hasnt actually fixed the original issue. The app turns blank after I scan a code in Release Builds.
Here is what I get in the logs

`10-30 15:04:01.319 12986 13045 I ReactNativeJS: '=================rawcodes=============>', { '0': undefined } 
 👆👆 👆This is the log that I added here I get an object of undefined. 

This are the ones that I get after that 👇👇👇

10-30 15:04:01.333 12986 13045 E ReactNativeJS: Error: Array out of bounds
10-30 15:04:01.333 12986 13045 E ReactNativeJS: 
10-30 15:04:01.333 12986 13045 E ReactNativeJS: This error is located at:
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in BarcodeScanner
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in StaticContainer
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in EnsureSingleNavigator
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in SceneView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in CardSheet
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in PanGestureHandler
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in PanGestureHandler
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Card
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in CardContainer
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RNSScreen
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Suspender
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Suspense
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Freeze
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in DelayedFreeze
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in InnerScreen
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Screen
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in MaybeScreen
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RNSScreenContainer
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in ScreenContainer
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in MaybeScreenContainer
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Background
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in CardStack
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RNCSafeAreaProvider
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in SafeAreaProvider
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in SafeAreaProviderCompat
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RNGestureHandlerRootView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in GestureHandlerRootView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in StackView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in PreventRemoveProvider
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in NavigationContent
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in StackNavigator
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in EnsureSingleNavigator
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in BaseNavigationContainer
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in ThemeProvider
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in NavigationContainerInner
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in AppNavigator
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in MainWrapper
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Provider
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Root
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in RCTView
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in Unknown
10-30 15:04:01.333 12986 13045 E ReactNativeJS:     in AppContainer, js engine: hermes
10-30 15:04:01.334 12986 13045 I HermesVM: JSI rethrowing JS exception: Array out of bounds
10-30 15:04:01.334 12986 13045 I HermesVM: `

**Comment by zzz08900** [comment]
Well, you can still try the other two methods and see if they work.

If you don't have a lot of fancy animations around I would recommend remove reanimated once anf for all -  from my experience it creates way too much trouble than it should as an animation lib.

**Comment by Nandha-d3v** [comment]
I noticed another issue that crashes the app (without worklets & frameprocessors. This happens when I try to scan the Barcodes multiple times. Also, I noticed this only in Release Builds.. 
The VisionCamera version is -  "react-native-vision-camera": "^3.6.4",

**Here are the logs**
`10-30 15:36:15.384 21610 21610 I CameraSession: Adding 1280 x 720 CodeScanner Output in Format #35...

--------- beginning of crash
10-30 15:36:15.384 21610 22086 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x7c6b882804 in tid 22086 (pool-6-thread-2), pid 21610 (com.myapp)
10-30 15:36:15.406 21610 21610 I CreateCaptureSession: Camera 0: Creating Capture Session #1041... Hardware Level: 3} | Outputs: [android.hardware.camera2.params.OutputConfiguration@c64c84f6]
10-30 15:36:15.406 21610 21610 I CreateCaptureSession: Using new API (>=28)
10-30 15:36:15.408 21610 21642 E BufferQueueProducer: [ImageReader-1280x720f23m2-21610-40](id:546a00000044,api:4,p:1803,c:21610) queueBuffer: BufferQueue has been abandoned
10-30 15:36:15.410 21610 22026 D TransportRuntime.SQLiteEventStore: Storing event with priority=DEFAULT, name=FIREBASE_ML_SDK for destination cct
10-30 15:36:15.412 21610 22026 D TransportRuntime.JobInfoScheduler: Upload for context TransportContext(cct, DEFAULT, MSRodHRwczovL2ZpcmViYXNlbG9nZ2luZy5nb29nbGVhcGlzLmNvbS92MGNjL2xvZy9iYXRjaD9mb3JtYXQ9anNvbl9wcm90bzNc) is already scheduled. Returning...
10-30 15:36:15.428 21610 21642 E BufferQueueProducer: [ImageReader-1280x720f23m2-21610-40](id:546a00000044,api:4,p:1803,c:21610) queueBuffer: BufferQueue has been abandoned
10-30 15:36:15.490 21610 21625 E BufferQueueProducer: [ImageReader-1280x720f23m2-21610-40](id:546a00000044,api:4,p:1803,c:21610) queueBuffer: BufferQueue has been abandoned
10-30 15:36:15.516 21610 21625 E BufferQueueProducer: [ImageReader-1280x720f23m2-21610-40](id:546a00000044,api:4,p:1803,c:21610) queueBuffer: BufferQueue has been abandoned
10-30 15:36:15.522 21610 21625 E BufferQueueProducer: [ImageReader-1280x720f23m2-21610-40](id:546a00000044,api:4,p:1803,c:21610) queueBuffer: BufferQueue has been abandoned
10-30 15:36:15.554 21610 21625 E BufferQueueProducer: [ImageReader-1280x720f23m2-21610-40](id:546a00000044,api:4,p:1803,c:21610) queueBuffer: BufferQueue has been abandoned
10-30 15:36:15.593 21610 21623 E BufferQueueProducer: [ImageReader-1280x720f23m2-21610-40](id:546a00000044,api:4,p:1803,c:21610) queueBuffer: BufferQueue has been abandoned
10-30 15:36:15.601 21610 21623 I CameraManagerGlobal: Camera 0 facing CAMERA_FACING_BACK state now CAMERA_STATE_IDLE for client com.nyapp API Level 2User Id 0`
